tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -63,8 +72,23 @@ void UpdateList(std::set<std::string> &bNamesReg, ColumnNames_t &bNames, std::st
       bNames.push_back(branchName);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// This overloads makes sure that the TLeaf has not been already inserted.
+void UpdateList(std::set<std::string> &bNamesReg, ColumnNames_t &bNames, std::string &branchName,
+                std::string &friendName, std::set<TLeaf *> &foundLeaves, TLeaf *leaf, bool allowDuplicates)
+{
+   const bool canAdd = allowDuplicates ? true : !ContainsLeaf(foundLeaves, leaf);
+   if (!canAdd) {
+      return;
+   }
+
+   UpdateList(bNamesReg, bNames, branchName, friendName);
+
+   foundLeaves.insert(leaf);
+}
+
 void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bNames, TBranch *b, std::string prefix,
-                   std::string &friendName)
+                   std::string &friendName, std::set<TLeaf *> &foundLeaves, bool allowDuplicates)
```

Comment:
I believe we do not need this extra params in `ExploreBranch` anymore.

Reply:
Fixed